src/ui/view/share/data/dropDown/dropDown.tsx

```diff
@@ -0,0 +1,125 @@
+export const DropDown: FC<DropDownProps> = ({
+  placeholder,
+  value,
+  options,
+  onChange,
+  searchPlaceholder,
+  selectionType,
+  maxSearchResults = APP_ENV.dropDownMaxSearchResults
+}) => {
+  const [searchTerm, setSearchTerm] = useState<string>('')
+
+  const handleSearch = useCallback(
+    (searchTerm: string) => {
+      setSearchTerm(searchTerm)
+    },
+    [setSearchTerm]
+  )
+
+  const handleSelectedChange = useCallback(
+    (v: string) => () => {
+      if (value.includes(v)) {
+        onChange(value.filter(val => val !== v))
+        return
+      }
+
+      onChange([...value, v])
+    },
+    [value, onChange]
+  )
+
+  const foundOptions = useMemo(
+    () =>
+      searchTerm.trim() === ''
+        ? options
+        : options.filter(v => isSubstringOf(searchTerm, v)).slice(0, maxSearchResults),
+    [options, searchTerm, maxSearchResults]
+  )
+
+  return (
+    <div className="okp4-dataverse-portal-dropdown-main">
+      <Collapsible
+        content={
+          <div className="okp4-dataverse-portal-dropdown-content">
+            <SearchBar onSearch={handleSearch} placeholder={searchPlaceholder} value={searchTerm} />
+            <div className="okp4-dataverse-portal-dropdown-options-list">
+              {foundOptions.length ? (
+                foundOptions.map(v => {
+                  switch (selectionType) {
+                    case 'checkbox':
+                      return (
+                        <DynamicCheckbox
+                          checked={value.includes(v)}
+                          highlightedTerm={searchTerm}
+                          key={v}
+                          name={v}
+                          onCheckedChange={handleSelectedChange(v)}
+                          value={v}
+                        />
+                      )
+                  }
+                })
+              ) : (
+                <NoResultFound
+                  className="okp4-dataverse-portal-dropdown-no-results-wrapper"
+                  iconName="large-magnifier-with-cross"
+                />
+              )}
+            </div>
+          </div>
+        }
+        iconName="chevron-sharp"
+        open
+        trigger={
+          <div className="okp4-dataverse-portal-dropdown-header">
+            {value.length ? (
+              <div className="okp4-dataverse-portal-dropdown-header-selection">
+                {value.slice(0, APP_ENV.dropDownTagsDisplayLimit).map(v => (
+                  <Tag
+                    classes={{ main: 'okp4-dataverse-portal-dropdown-header-selection-item' }}
+                    key={v}
+                    onDelete={handleSelectedChange(v)}
+                    tagName={v}
+                  />
+                ))}
+                {value.length > APP_ENV.dropDownTagsDisplayLimit && (
+                  <Tag
+                    classes={{ main: 'okp4-dataverse-portal-dropdown-header-selection-item' }}
+                    tagName={`+${String(value.length)}`}
```

Comment:
You need to subtract default display limit. Also, string casting seems unnecessary.
```suggestion
                    tagName={`+${value.length - dropDownTagsDisplayLimit}`}
```

Reply:
That's what I taught too but when I read the spec, "... a tag containing the total number of selected options appears." I get that it should be the total of selected items

Reply:
It seems counterintuitive with the leading `+`. We probably need a confirmation from the product team.

Reply:
@0xLodz I've adressed this issue with them, they agree that we both taught is the right implementation